plugins/core/qPCV/PCVCommand.cpp

```diff
@@ -0,0 +1,101 @@
+#include "PCVCommand.h"
+#include "PCV.h"
+#include "qPCV.h"
+
+//qCC_db
+#include <ccGenericMesh.h>
+#include <ccHObjectCaster.h>
+#include <ccPointCloud.h>
+#include <ccProgressDialog.h>
+
+static const char COMMAND_PCV[] = "PCV";
+static const char COMMAND_PCV_N_RAYS[] = "N_RAYS";
+static const char COMMAND_PCV_IS_CLOSED[] = "IS_CLOSED";
+static const char COMMAND_PCV_180[] = "180";
+static const char COMMAND_PCV_RESOLUTION[] = "RESOLUTION";
+
+PCVCommand::PCVCommand() :
+	Command("PCV", COMMAND_PCV)
+{
+}
+
+bool PCVCommand::process(ccCommandLineInterface& cmd)
+{
+	cmd.print("[PCV]");
+
+	// Initialize to match PCV::Launch defaults
+	unsigned nRays = 256;
+	bool meshIsClosed = false;
+	bool mode360 = true;
+	unsigned resolution = 1024;
+
+	while (!cmd.arguments().empty())
+	{
+		QString arg = cmd.arguments().front();
+		if (ccCommandLineInterface::IsCommand(arg, COMMAND_PCV_IS_CLOSED))
+		{
+			cmd.arguments().pop_front();
+			meshIsClosed = true;
+		}
+
+		// PCV::Launch mode360 defaults to true. To make absence / presence of command map to false / true,
+		// the command is named "180," and is the inverse of mode360.
+		else if (ccCommandLineInterface::IsCommand(arg, COMMAND_PCV_180))
+		{
+			cmd.arguments().pop_front();
+			mode360 = false;
+		}
+		else if (ccCommandLineInterface::IsCommand(arg, COMMAND_PCV_N_RAYS))
+		{
+			cmd.arguments().pop_front();
+			bool conversionOk = false;
+			nRays = cmd.arguments().takeFirst().toFloat(&conversionOk);
+			if (!conversionOk)
+			{
+				return cmd.error(QObject::tr("Invalid parameter: value after \"-%1\"").arg(COMMAND_PCV_N_RAYS));
```

Comment:
Where you are using `QObject::tr()`, please use `qPCV::tr()` so the translations are grouped properly for the translators.

Reply:
Addressed in df3b2f59dada2391d69afcacf88e0154701c5061

Reply:
Re-addressed in 545a8db082f63efe2d72f32d86b4b1640babaffe -- I realized two of these strings are not qPCV specific -- they're used in other CLI commands. One of them (an implementation detail) is. There was a fourth string, which reported the number of points in the point cloud, but the progress bar window also displays that, so I removed it since it seems like unnecessary duplication.